### PR TITLE
docs(socket-engine): Update README

### PR DESCRIPTION
**Pull request by AlexBachmann**
Update the README of the socket-engine implementation to correctly show, how to setup a socket-server, using Angular 9. Flesh out the guide with more details on the setup process. Highlight common pitfalls with localized apps.

### CODE REVIEW

File: modules/socket-engine/README.md [hunk 1]
```diff
@@ -2,21 +2,63 @@
...
 `npm install @nguniversal/socket-engine @nguniversal/common --save`
 
+Now replace the **server.ts** node express implementation, with the following code:
+
 ```js
 const socketEngine = require('@nguniversal/socket-engine');
 
 // * NOTE :: leave this as require() since this file is built Dynamically from webpack
-const {AppServerModuleNgFactory, LAZY_MODULE_MAP} = require('./dist/server/main');
+const {AppServerModule} = require('./src/main.server');
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: Can we update this code here to reflect the latest server.ts?

File: modules/socket-engine/README.md [hunk 2]
```diff
@@ -2,21 +2,63 @@
...
+    |-- main.ts
+    |-- **main.server.ts**
+|-- **server.ts**
+|-- tsconfig.app.json
+|-- tsconfig.json
+|-- **tsconfig.server.json**
+
+To switch from the node express server implementation to the socket engine implementation you first need to install two additional dependencies. 
 
 `npm install @nguniversal/socket-engine @nguniversal/common --save`
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: This is actually a good feature request to an ng add for this package.

File: modules/socket-engine/README.md (hunk 1 above)
Comment (AlexBachmann) [reply modules/socket-engine/README.md]: I did. It looks much better now.

File: modules/socket-engine/README.md [hunk 3]
```diff
@@ -2,21 +2,64 @@
...
+// import 'zone.js/dist/zone-node';
+import { startSocketEngine } from '@nguniversal/socket-engine';
+import { AppServerModule } from './src/main.server';
 
-// * NOTE :: leave this as require() since this file is built Dynamically from webpack
-const {AppServerModuleNgFactory, LAZY_MODULE_MAP} = require('./dist/server/main');
+const port: number = parseInt(process.env.PORT, 10) || 9090;
 
-socketEngine.startSocketEngine(AppServerModuleNgFactory);
+console.log('Starting the socket-server on port: ' + port);
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: Nit: `	console.log(`Starting the socket-server on port: ${port}`);`

File: modules/socket-engine/README.md [hunk 4]
```diff
@@ -2,21 +2,64 @@
...
+To switch from the node express server implementation to the socket engine implementation you first need to make sure that the two dependencies, the socket engine requires, are installed. 
 
 `npm install @nguniversal/socket-engine @nguniversal/common --save`
 
-```js
-const socketEngine = require('@nguniversal/socket-engine');
+Now replace the **server.ts** node express implementation, with the following code:
+
+```typescript
+// You proboably won't need zone.js for the server compilation, as long as you compile for production
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: Why Zone-JS is not needed?

File: modules/socket-engine/README.md [hunk 5]
```diff
@@ -2,21 +2,64 @@
...
+This will start the socket engine which internally hosts a TCP Socket server.  
+The default port is `9090` and host of `localhost`.
+
+## Compile your app
+
+After everything is setup, it's time to compile the app. You can simply run
+
+`npm run build:ssr && npm run serve:ssr`
+
+`npm run build:ssr` is a shortcut for running `ng build --prod && ng run app:server:production`, which first compiles the normal browser app and then compiles the server app, based on that output.
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: Nit: can we use full terms please `app` -> `application`

File: modules/socket-engine/README.md [hunk 6]
```diff
@@ -2,21 +2,64 @@
...
+
+`npm run serve:ssr` is a shortcut for running `node dist/app/server/main.js`. 
+
+## Notes for Localization (i18n)
+
+If you're using the [--localize](https://angular.io/guide/i18n) option while building your app, you might not be able to use the `npm run serve:ssr` shortcut, since the compiler generates one main.js bundle for each language you compile (the compiled server app might be located in `node dist/app/server/de-DE/main.js` for example).
+
+Furthermore, you need to be aware that you need to enable localization for the compilation of the browser app AND the server app. You can do that by using the `--localize` flag or by setting `"localize": true"` in the in your `angular.json` file (within the `options` namespace).
```
Comment (alan-agius4) [inline modules/socket-engine/README.md]: Same as previous

**Comment by CaerusKaru** [comment]
@AlexBachmann Are you still interesting in finishing this? If not, we can reassign it to someone else on the team to close out.

**Comment by alan-agius4** [comment]
Closing since PR is stale.

**Pull request closed without merging**